### PR TITLE
Feature: replace ibc-hook with middleware

### PR DESCRIPTION
**Pull request by thomas-nguy**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by tomtau — COMMENTED** [review]
🎉awesome.
I guess the problem statement here is: `Problem: custom ibc-go fork used`? the docs can perhaps have a brief update: https://github.com/crypto-org-chain/cronos/blob/main/docs/architecture/adr-002.md and https://github.com/crypto-org-chain/cronos/blob/main/docs/architecture/adr-004.md that the middleware implementation will override `OnRecvPacket`, `OnAcknowledgmentPacket`, and `OnTimeoutPacket`?

One potential thing to check is that there's a missing test for timeouts etc.: https://github.com/crypto-org-chain/cronos/issues/398 -- we can see where it's at with the QA team... if ready soon, maybe we can merge the missing integration test first and run it against this?

File: x/cronos/types/interfaces.go [hunk 1]
```diff
@@ -76,3 +79,10 @@ type EvmKeeper interface {
 	) (sdk.Coins, error)
 	ChainID() *big.Int
 }
+
+// ICS4Wrapper defines the expected ICS4Wrapper for middleware
+type ICS4Wrapper interface {
```
Comment (yihuang) [inline x/cronos/types/interfaces.go:84]: this seems not used anywhere?

Comment (thomas-nguy) [reply x/cronos/types/interfaces.go:84]: yes this can be removed for now (this is needed if we need to write a "full" middleware)

**Review by yihuang — APPROVED** [review]
LGTM

**Comment by thomas-nguy** [comment]
I will merge this PR first as the current integration tests are passing.

I will work on the integration tests for ibc timeout/failure in another PR 
https://github.com/crypto-org-chain/cronos/issues/514

**Review by adu-crypto — APPROVED** [review]
lgtm

**Pull request merged**